google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/vcs/CloneGcpDialog.java

```diff
@@ -150,51 +150,77 @@ protected String getHelpId() {
    * Check fields and display error in the wrapper if there is a problem.
    */
   private void updateButtons() {
-    if (parentDirectory.getText().length() == 0 || directoryName.getText().length() == 0) {
+    if (!StringUtil.isEmpty(projectSelector.getText())
+        && projectSelector.getSelectedUser() == null) {
+      setErrorText(GctBundle.message("cloud.repository.dialog.invalid.project"));
+      setOKActionEnabled(false);
+      return;
+    }
+
+    if (!StringUtil.isEmpty(repositorySelector.getText())
+        && StringUtil.isEmpty(repositorySelector.getSelectedRepository())) {
+      setErrorText(GctBundle.message("cloud.repository.dialog.invalid.repository"));
+      setOKActionEnabled(false);
+      return;
+    }
+
+    if(projectSelector.getSelectedUser() == null
+        || StringUtil.isEmpty(repositorySelector.getSelectedRepository())) {
+      setErrorText(null);
+      setOKActionEnabled(false);
+      return;
+    }
+
+    if (StringUtil.isEmpty(parentDirectory.getText())
+        || StringUtil.isEmpty(directoryName.getText())) {
       setErrorText(null);
       setOKActionEnabled(false);
       return;
     }
+
     File file = new File(parentDirectory.getText(), directoryName.getText());
```

Comment:
Would it make sense to refactor here to use java.nio? It would come with the annoyance of having to catch InvalidPathExceptions from paths with special characters on Windows...